packages/next/src/server/lib/incremental-cache/fetch-cache.ts

```diff
@@ -168,8 +168,17 @@ export default class FetchCache implements CacheHandler {
     // on successive requests
     let data = memoryCache?.get(key)
 
-    // get data from fetch cache
-    if (!data && this.cacheEndpoint) {
+    const isCacheable = !data && this.cacheEndpoint
+    const hasFetchKindAndIncludesAllTags = tags?.every((tag) => {
+      if (data?.value?.kind !== 'FETCH') {
+        return false
+      }
+      return data.value.tags?.includes(tag)
+    })
+
+    // Get data from fetch cache. Also check if new tags have been
+    // specified with the same cache key (fetch URL)
+    if (isCacheable || !hasFetchKindAndIncludesAllTags) {
```

Comment:
```suggestion
    if (this.cacheEndpoint && !memoryCacheMatchesTags) {
```
If we don't have a memory cache entry or the entry doesn't match all tags we need to hit upstream

Reply:
this makes that variable `isCacheable` unused. Should this check be `!data && this.cacheEndpoint && !memoryCacheMatchesTags`? Or will the `!memoryCacheMatchesTags` cover the `!data` part 🤔